### ISSUE TITLE
Render Issues when trying to send looks including viusualizations to Slack

### ISSUE THREAD

**Issue by kobyys** [looker/lookerbot#148]
When trying to schedule or send a look to lookerbot containing visualizations I get the error message ":warning: An error has occurred." in Slack. Sending looks without visualizations works fine. I run a self-hosted Instance of looker.

The error I receive in the looker.log, when sending a look containing a visualization is the following:

```
StandardError` : Render job 5e448255f095a2a04279091ff588fc0c[5882] failed ["Unexpected document content"]
```

The lookerbot documentation said the format selection for the output sent to the webhook would be ignored. However when I analysed the traffic sent to lookerbot, I saw that the content sent to lookerbot ss always the format selection I chose, when scheduling the look. For example CSV, html and so on.

I did set environment variables for a google cloud bucket. However the error appears regardless of these variables being set.

I have tried to give the bot user all different kind of rights in looker and tested all options through sudoing as the bot user. This should not be related to looker rights.

I am currently running looker Version 5.16.18

Can someone help me with this or at least tell me which format looker should use to send data containing a visualization to lookerbot?

**Comment by wilg** [comment]
I think the documentation is still correct – due to limitations in how data could be sent at the time Lookerbot actually runs the query twice.

It looks like an error on Looker's end, I'll do some investigation internally.

**Comment by wilg** [comment]
Does downloading PDFs or images from the Looker interface work properly for you?

**Comment by kobyys** [comment]
> Does downloading PDFs or images from the Looker interface work properly for you?
Yes I can download PDFs and images correctly, as well as send them out via mail.

**Comment by kobyys** [comment]
I now tried it with a different google cloud bucket in a different project with a different credentials JSON file.
The result was still the same. The error still looks like this:

```
2018-09-14 08:40:10.066 +0000 [ERROR|033a2|] :: StandardError : Render job c85ac6e7dba04d5d29d0685d8c7521b0[6498] failed ["Unexpected document content"]
2018-09-14 08:40:48.286 +0000 [ERROR|033da|] :: StandardError : Render job b7b44d462607a3e19ddd3c08e110ed15[6499] failed ["Unexpected document content"]
2018-09-14 08:41:01.518 +0000 [ERROR|033f6|] :: StandardError : Render job d02aeb20006e95bfe0478868406ee633[6500] failed ["Unexpected document content"]
```

These are the only errors listed in `looker.log`

I will update from looker Version 5.16.18 to the current version soon, maybe that will help the problem.

**Comment by kobyys** [comment]
Hello,

I updated looker to `5.24` and fortunately the error message improved.
When I am sending data to slack now, I receive the following error in slack:

`SyntaxError: Unexpected token / in JSON at position 0`

@wilg Could this potentially help you isolating what is going wrong?

Thank you for your efforts!

**Comment by kobyys** [comment]
I think I realized what is going on. I think something changed about labs features and the documentation.
If I recall correctly the documentation used to say 
``` 
Certain scheduling options require that Looker admins have enabled the Enhanced Rendering Labs feature and that on-premise Looker admins have installed PhantomJS for their Looker instance.
```
Now it is:
```
Certain scheduling options require that Looker admins have enabled the Enhanced Rendering Labs feature and that on-premise Looker admins have installed PhantomJS or Chromium for their Looker instance
```
Furthermore an option I was looking for, for the longest time, in Labs features appeared with the latest version called `Enhanced Rendering`. Even though the documentation always linked to this option, for what I was trying to achieve. it never showed up on my on-premise-Instance. When I try to activate it now, however it gives the following output:
`Chromium 65 or above must be installed on the server to enable this feature.`

I assume this option never showed up before, because I did not have Chromium installed on the machine. Sadly my looker-Instance is running on Debian8 and I cannot install Chromium 65 on it, because of that for now.

On another note a further documentation page is now available, or at least I did not ever find it before.
It is the following: https://docs.looker.com/setup-and-management/on-prem-install/on-prem-labs

It lists the necessity of Chromium 65.

Even though my problem is not gone, I know why it won't work through the added documentation.

**most likely solution: Update to Debian 9.**

I guess this Issue can be closed.
